source/DataReader.cs

```diff
@@ -285,7 +272,9 @@ public float ReadSingle()
         /// <returns>The value.</returns>
         public string ReadString(UInt32 codeUnitCount)
         {
-            var value = new string(Encoding.UTF8.GetChars(_buffer.Data, IncreaseReadPosition((int)codeUnitCount), (int)codeUnitCount));
+            var _chars = new Char[_buffer.Data.Length];
```

Comment:
Could you please follow the project C# coding style and rename the local var by removng the starting underscore? That's reserved for class internal and private fields. 😉